test/integration/src/test/java/io/pravega/test/integration/LargeEventTest.java

```diff
@@ -0,0 +1,523 @@
+        for (int i = 0; i < writers; i++) {
+            log.info("Starting Writer {}", i);
+            String routingKey = String.format("LargeEventWriter-%d", i);
+            int writerId = i;
+            Runnable restart = () -> {
+                log.info("Closing writer {} ClientConnection.", writerId);
+                exporter.getConnection().close();
+            };
+            writerList.add(startNewWriterPreflushAction(routingKey, streamName, eventWriteCount, eventsWrittenToPravega.get(i), factory, restart));
+        }
+        return writerList;
+    }
+
+    private void createReaderGroup(String group, ReaderGroupManager manager, String streamName) {
+        log.info("Creating ReaderGroup : {}", group);
+        manager.createReaderGroup(group, ReaderGroupConfig.builder().stream(Stream.of(SCOPE_NAME, streamName)).build());
+        log.info("ReaderGroup Name : {} ", manager.getReaderGroup(group).getGroupName());
+        log.info("ReaderGroup Scope : {}", manager.getReaderGroup(group).getScope());
+    }
+
+    private List<CompletableFuture<Void>> createEventReaders(int readers, ClientFactoryImpl factory, String readerGroupName) {
+        log.info("Creating {} Readers.", readers);
+        List<CompletableFuture<Void>> readerList = new ArrayList<>();
+        for (int i = 0; i < readers; i++) {
+            String readerId = String.format("LargeEventReader-%d", i);
+            readerList.add(startNewReader(readerId,
+                    factory,
+                    readerGroupName,
+                    eventsReadFromPravega,
+                    eventWriteCount,
+                    eventReadCount,
+                    stopReadFlag));
+        }
+        return readerList;
+    }
+
+    public void validateCleanUp(String streamName) throws ExecutionException, InterruptedException {
+        // Seal the stream
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(SCOPE_NAME, streamName);
+        log.info("Sealing stream {}", streamName);
+        assertTrue(sealStreamStatus.get());
+        assertTrue(sealStreamStatus.get());
+
+        //delete the stream
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(SCOPE_NAME, streamName);
+        log.info("Deleting stream '{}'", streamName);
+        assertTrue(deleteStreamStatus.get());
+        //delete the  SCOPE_NAME
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(SCOPE_NAME);
+        log.info("Deleting scope '{}'", SCOPE_NAME);
+        assertTrue(deleteScopeStatus.get());
+    }
+
+    @Test
+    public void readWriteTest() throws ExecutionException, InterruptedException {
+        String readerGroupName = "testLargeEventReaderGroup";
+        String streamName = "ReadWrite";
+        StreamConfiguration config = getStreamConfiguraton(NUM_READERS);
+        createScopeStream(SCOPE_NAME, streamName, config);
+        generateWriteEventData(NUM_WRITERS, Serializer.MAX_EVENT_SIZE * 5);
```

Comment:
Tip: let's put `Serializer.MAX_EVENT_SIZE * 5` into a class-level constant so that we may easily change it later if we have to.